python-checks/src/main/java/org/sonar/python/checks/HardcodedIPCheck.java

```diff
@@ -68,15 +68,34 @@ public void visitNode(AstNode node) {
       matcher = patternV6.matcher(string);
       if (matcher.find()) {
         String ipAddress = matcher.group();
-        if (ipAddress.length() > 8) {
+        if (ipAddress.length() > 8 && !mustBeExcluded(string, ipAddress)) {
           addIssue(node, String.format(message, ipAddress));
         }
       }
     }
   }
 
+  /**
+   * Returns true if the match is preceded by or followed by a letter or a number.
+   */
+  private static boolean mustBeExcluded(String string, String ipAddress) {
+    // check the character before the match
+    int index = string.indexOf(ipAddress);
+    if (string.substring(index - 1, index).matches("[\\da-fA-F]")) {
+      return true;
+    }
+
+    // check the character after the match
+    index += ipAddress.length();
+    if (string.substring(index, index + 1).matches("[\\da-fA-F]")) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1126)